api/misc.go

```diff
@@ -29,18 +31,65 @@ func (s *Server) eventList(apiContext *api.ApiContext) (*client.GenericCollectio
 	return toEventCollection(eventList), nil
 }
 
-func (s *Server) GenerateSupportBundle(w http.ResponseWriter, req *http.Request) error {
-	bundleFile, fileName, size, err := s.m.GenerateSupportBundle()
+func (s *Server) InitiateSupportBundle(w http.ResponseWriter, req *http.Request) error {
+	var sb *manager.SupportBundle
+	apiContext := api.GetApiContext(req)
+	sb, err := s.m.InitSupportBundle()
 	if err != nil {
+		return errors.Errorf("unable to initiate Support Bundle Download:%v", err)
+	}
+	if sb != nil {
+		apiContext.Write(toSupportBundleResource(s.m.GetCurrentNodeID(), sb))
+	}
+	return nil
+}
+
+func (s *Server) QuerySupportBundle(w http.ResponseWriter, req *http.Request) error {
+	var sbFile SupportBundleQueryInput
+	apiContext := api.GetApiContext(req)
+	if err := apiContext.Read(&sbFile); err != nil {
+		return err
+	}
+	sb := s.m.GetSupportBundleObject()
+	if strings.Compare(sbFile.Filename, sb.Fname) != 0 {
+		return errors.Errorf("cannot find the file:%v", sbFile.Filename)
+	}
+	if sb.State == manager.BundleStateError {
+		err := errors.Errorf("support bundle creation failed:%s", sb.Error)
```

Comment:
Don't need to return an error in this case. The client will receive the error in the bundle state and error message.